modules/gateway/persist.go

```diff
@@ -78,3 +76,22 @@ func (g *Gateway) threadedSaveLoop() {
 		}()
 	}
 }
+
+// loadv033persist loads the v0.3.3 Gateway's persistent data from disk.
+func (g *Gateway) loadv033persist() error {
+	var nodes []modules.NetAddress
+	err := persist.LoadJSON(persist.Metadata{
+		Header:  "Sia Node List",
+		Version: "0.3.3",
+	}, &nodes, filepath.Join(g.persistDir, nodesFile))
+	if err != nil {
+		return err
+	}
+	for _, addr := range nodes {
+		err := g.addNode(addr)
+		if err != nil {
+			g.log.Printf("WARN: error loading node '%v' from persist: %v", addr, err)
+		}
+	}
+	return nil
+}
```

Comment:
Added in https://github.com/NebulousLabs/Sia/commit/de43b04b95e8cee1e9e3b5d49c2472455b9c839a